src/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -125,5 +114,64 @@ public static IEnumerable<string> GetAllProjectFileNames(string solutionFile)
                 throw new CommandLineException(message);
             }
         }
+
+        public static string GetMsbuildDirectory(string version)
+        {
+            if (String.IsNullOrEmpty(version))
+            {
+                // default to 4.0
+                version = "4.0";
+
+                // run msbuild to get the version
+                var processStartInfo = new ProcessStartInfo
+                {
+                    UseShellExecute = false,
+                    FileName = "msbuild.exe",
+                    Arguments = "/version",
+                    RedirectStandardOutput = true
+                };
+                try
+                {
+                    using (var process = Process.Start(processStartInfo))
+                    {
+                        process.WaitForExit(100);
```

Comment:
I'm not sure if 100ms will be enough time on a build machine running several builds in parallel.